### ISSUE TITLE
build failure with gcc14 src/burner/unzip.c:1182:24: error: assignment to ‘const long unsigned int

### ISSUE THREAD

**Issue by jeandestouches** [libretro/fbalpha2012#116]
Hello,
Build failure with GCC14 (no issue with GCC12 using same commit) on my gentoo system.

Log:

x86_64-pc-linux-gnu-gcc -march=bdver2 -O2 -pipe -march=bdver2 -O2 -pipe -DNDEBUG -fPIC -Wno-write-strings -DGIT_VERSION=\"" b7ac554"\" -DINCLUDE_7Z_SUPPORT -DUSE_SPEEDHACKS -D__LIBRETRO__ -D__LIBRETRO_OPTIMIZATIONS__ -DWANT_NEOGEOCD   -D__LIBRETRO_OPTIMIZATIONS__ -DFRONTEND_SUPPORTS_RGB565 -c -osrc/burn/snd/ym2151.o src/burn/snd/ym2151.c -march=bdver2 -O2 -pipe -march=bdver2 -O2 -pipe -DNDEBUG -fPIC -Wno-write-strings -DGIT_VERSION=\"" b7ac554"\" -DINCLUDE_7Z_SUPPORT -DUSE_SPEEDHACKS -D__LIBRETRO__ -D__LIBRETRO_OPTIMIZATIONS__ -DWANT_NEOGEOCD   -D__LIBRETRO_OPTIMIZATIONS__ -DFRONTEND_SUPPORTS_RGB565 -Isrc/burner/libretro -Isrc/burn -Isrc/cpu -Isrc/burn/snd -Isrc/burn/devices -Isrc/intf -Isrc/intf/input -Isrc/intf/cd -Isrc/burner -Isrc/cpu -Isrc/cpu/i8039 -Isrc/dep/libs/zlib -Isrc/dep/libs/lib7z -Isrc/burn/drv/capcom -Isrc/burn/drv/dataeast -Isrc/burn/drv/cave -Isrc/burn/drv/neogeo -Isrc/burn/drv/psikyo -Isrc/burn/drv/sega -Isrc/burn/drv/toaplan -Isrc/burn/drv/taito -Isrc/dep/generated -Isrc/burner/libretro/libretro-common/include -Isrc/dep/libs
src/burner/unzip.c: In function ‘unzOpenCurrentFile3’:
src/burner/unzip.c:1182:24: error: assignment to ‘const long unsigned int *’ from incompatible pointer type ‘const z_crc_t *’ {aka ‘const unsigned int *’} [-Wincompatible-pointer-types]
 1182 |         s->pcrc_32_tab = get_crc_table();
      |                        ^
make: *** [makefile.libretro:806: src/burner/unzip.o] Error 1
make: *** Waiting for unfinished jobs....
 * ERROR: games-emulation/libretro-fbalpha2012-9999-r1::localrepo failed (compile phase):
 *   emake failed

**Comment by RogueScholar** [comment]
Yeah, this isn't just you, there's definitely some major code smells here. This core always compiled smooth as silk for the Launchpad PPAs for eons and then with the release of Ubuntu 24.10 "Oracular Oriole" that has GCC 14 as the default C++ compiler, it crashes out with the exact same error as you get on like the very first compilation unit for every architecture we build on.

How do they keep the nightly cores being built, just by never upgrading the compiler toolchain? I'd rather they simply withdraw support for cores when there's no interest in maintaining them, myself. Anyhow, I wanted to validate your experience since this issue has been in my watchlist for awhile and nobody's responded to it, still. Cheers.
